### ISSUE TITLE
Suite options with extra spaces don't produce a warning

### ISSUE THREAD

**Issue by ttusar** [numbbo/coco#863]
If, for example, suite options are written like this:
`"dimensions: 2, 3, 5, 10 function_indices: 20"`
The experiments will be run only for dimension 2 without any warning.

Issue first mentioned in #724

**Comment by nikohansen** [comment]
I would consider this as a bug as long as not even a warning is produced.

**Comment by brockho** [comment]
Seems to work most of the times, indeed.

The above example also works for me as does `"dimensions: 2, 3, 5, 10 function_indices: 20, 21"`:
```
COCO INFO: Results will be output to folder exdata\random_search_on_bbob_budget0001xD-007
Benchmarking solver '<function random_search' with budget=1*dimension on bbob suite, Thu Aug 31 17:4
6:57 2017
Aug31 17h46:57, d=2, running: f020...............(15)  f021...............(15)
    done in 6.0e-03 seconds/evaluation

Aug31 17h46:58, d=3, running: f020...............(15)  f021...............(15)
    done in 3.9e-03 seconds/evaluation

Aug31 17h46:58, d=5, running: f020...............(15)  f021...............(15)
    done in 2.8e-03 seconds/evaluation

Aug31 17h46:59, d=10, running: f020...............(15)  f021...............(15)
    done in 1.4e-03 seconds/evaluation
  dimension seconds/evaluations
  -----------------------------
      2      6.0e-03
      3      3.9e-03
      5      2.8e-03
     10      1.4e-03
  -----------------------------
  bbob done (120 of 120 problems benchmarked), Thu Aug 31 17:46:59 2017 (0h00:01 total elapsed time)
```

However if I run it with `"dimensions: 2, 3, 5, 10 function_indices: 20"` the output is not showing the `f20` in all but dimension 2 and summarizes the total number of instances run wrongly:
```
COCO INFO: Results will be output to folder exdata\random_search_on_bbob_budget0001xD-005
Benchmarking solver '<function random_search' with budget=1*dimension on bbob suite, Thu Aug 31 17:4
5:52 2017
Aug31 17h45:52, d=2, running: f020...............
    done in 5.3e-03 seconds/evaluation

Aug31 17h45:53, d=3, running: ...............
    done in 3.9e-03 seconds/evaluation

Aug31 17h45:53, d=5, running: ...............
    done in 2.6e-03 seconds/evaluation

Aug31 17h45:53, d=10, running: ...............(60)
    done in 1.7e-03 seconds/evaluation
  dimension seconds/evaluations
  -----------------------------
      2      5.3e-03
      3      3.9e-03
      5      2.6e-03
     10      1.7e-03
  -----------------------------
  bbob done (60 of 60 problems benchmarked), Thu Aug 31 17:45:53 2017 (0h00:00 total elapsed time).
```

But that is an error that must have been there before because already ""dimensions: 2,3,5,10 function_indices: 20"" (without any wrongly placed blanks") gives the same output.

Note that the output in the `exdata` folder looks okay in the above cases.

**Comment by islandblue** [comment]
@brockho, can you also post here what code you ran to get that output? I'd like to step through it and see if I can trace the culprit, or at least run the same thing on the development branch to see if it's caused by this commit. Because if it isn't, I'd much prefer to let the pull request go through and write this up as a new issue.

**Comment by brockho** [comment]
@Umut: I simply ran the `example_experiment.py` in `code-experiments/build/python/` after `python do.py build-python` and changed the suite options in line 376 to the above, i.e. for example to
```
suite_options = "dimensions: 2, 3, 5, 10 function_indices: 20"
```

**Comment by nikohansen** [comment]
The wrong output summary is most likely a bug in the class `ShortInfo` of `example_experiment.py`.

**Comment by islandblue** [comment]
Probable reason for `f020` getting printed only in the first run:

https://github.com/numbbo/coco/blob/4c68f2f26a642ae4b772b818351afe49a4da9b4e/code-experiments/build/python/example_experiment.py#L201-L203

It seems to only print out when function changes, and when there is one function, it doesn't really change.

**Comment by islandblue** [comment]
Likewise, for (60) printing only once at the end...

https://github.com/numbbo/coco/blob/4c68f2f26a642ae4b772b818351afe49a4da9b4e/code-experiments/build/python/example_experiment.py#L195-L196

**Comment by islandblue** [comment]
Certain outputs seem to be triggered by a change in the function. This is obviously an oversight, as demonstrated. The function need not change.

I'm not 100% comfortable with the subject matter, but I think we relied on there being more than one function in each dimension.

**Comment by nikohansen** [comment]
changing Line 195 to
```python
if self.f_current and (f != self.f_current or problem.dimension != self.d_current):
```
may do the job. This code is only related to printing progress reports on the console, so changes won't break any other functionality.

**Comment by brockho** [comment]
The code solving this issue was just merged.
